string name
string goal
string text
